src/Grpc.AspNetCore.Server/Internal/GrpcProtocolHelpers.cs

```diff
@@ -27,6 +32,9 @@ namespace Grpc.AspNetCore.Server.Internal
 {
     internal static class GrpcProtocolHelpers
     {
+        // https://stackoverflow.com/a/54434121/11829
+        private static readonly Regex SubjectAlternativeNameRegex = new Regex("((DNS:|DNS Name=|CN=)(?<domain>.*?),?(?=\\s|$))+", RegexOptions.CultureInvariant, TimeSpan.FromSeconds(1));
```

Comment:
@blowdard @javiercn do you have expertise on how X509Certificate2 works? Or would you be able to check whether the logic here is correct?

Reply:
The unit tests now passing on my computer (Windows) and the CI (Linux) confirms it works. I searched and there is no alternative API for this with X509Certificate2.

Reply:
@blowdart

Reply:
Ummmmmm no regexes please. What are you trying to do here?

Reply:
As far as I can tell the only way to access alternative names on X509Certificate2 is to call https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.asnencodeddata.format?view=netframework-4.7.2#System_Security_Cryptography_AsnEncodedData_Format_System_Boolean_

But there are slight differences in the string result depending on the operating system:
https://stackoverflow.com/questions/54358492/cant-read-certificate-when-running-in-linux-docker-container-works-on-windows/54434121#54434121

The regex handles paring those differences.

Is there a better solution?

Reply:
Ugh. Forgot that.

Does this indicate that GRPC does mistreats certificates for auth in general though? (Multiple subjects is weird here)

Reply:
I don't know. We (gRPC for ASP.NET) don't use AuthContext ourselves. It is only provided for compatibility with existing gRPC apps that port from the other .NET implementation.

How people use AuthContext is up to them.

Reply:
You make me cry. Is AuthContext from the gRPC spec, or is it just from other implementations? If it's the latter then frankly I would rather make some changes to be more secure, and have a "UseCrappyCertParsing" as an option.

Reply:
It is from other implementations.